build-tools/garbage-files-generator.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import math
+import os
+import pathlib
+import re
+from functools import reduce
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(levelname)-8s - %(filename)s:%(lineno)d |> %(message)s",
+)
+
+try:
+    import tqdm
+except ImportError:
+    tqdm = None
+    logging.warning(
+        f"tqdm library is not available, progress bar will not be displayed."
+    )
+
+
+OUTPUT_FOLDER = pathlib.Path("./data")
+
+
+def main():
+    args = _parse_args()
+    generate_data(
+        args.total_size,
+        args.files_count,
+        args.branching_factor,
+    )
+
+
+def generate_data(
+    total_size: int,
+    files_count: int,
+    branching_factor: int,
+):
+    file_size_bytes = math.ceil(total_size / files_count)
+    tree_depth = math.floor(math.log(files_count, branching_factor))
+    name_length = len(str(branching_factor))
+    logging.info(
+        f"Generating {files_count} files {file_size_bytes} bytes each, "
+        f"which in total: {total_size} bytes."
+    )
+
+    OUTPUT_FOLDER.mkdir(parents=True, exist_ok=True)
+    fs_stats = os.statvfs(OUTPUT_FOLDER)
+    if file_size_bytes < 10 * fs_stats.f_frsize:
+        resulting_size = (
```

Comment:
Ok, added just in case if someone will be curious why the generated dataset size is larger than requested. But in general, you're right, it will not impact the generation performance, removed in 4955b95